src/System.IO/tests/Stream/Stream.NullTests.cs

```diff
@@ -5,31 +5,91 @@
+        [Theory]
+        [MemberData(nameof(NullStream_CopyToData))]
+        public async static Task TestNullStream_CopyTo(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            Stream.Null.CopyTo(destination, bufferSize);
+            await Stream.Null.CopyToAsync(destination, bufferSize, cancellationToken);
+        }
+        
         [Fact]
-        public static async Task TestNullStream()
+        public async static Task TestNullStream_CopyToAsyncValidation()
         {
-            Stream s = Stream.Null;
-
-            s.Flush();
-
-            Assert.Equal(-1, s.ReadByte());
-
-            s.WriteByte(5);
-            int n = s.Read(new byte[2], 0, 2);
-            Assert.Equal(0, n);
-            s.Write(new byte[2], 0, 2);
-
-            Assert.Equal(0, await s.ReadAsync(new byte[2], 0, 2));
-
-            s.Flush();
-            s.Dispose();
+            // Since Stream.Null previously inherited its CopyToAsync
+            // implementation from the base class, which did check its
+            // arguments, we have to make sure it validates them as
+            // well for compat.
+            
+            await Assert.ThrowsAsync<ArgumentNullException>("destination", () => Stream.Null.CopyToAsync(null));
+            await Assert.ThrowsAsync<ArgumentNullException>("destination", () => Stream.Null.CopyToAsync(null, -123)); // Should check if destination == null first
+            await Assert.ThrowsAsync<ArgumentOutOfRangeException>("bufferSize", () => Stream.Null.CopyToAsync(Stream.Null, 0)); // 0 shouldn't be a valid buffer size
+            await Assert.ThrowsAsync<ArgumentOutOfRangeException>("bufferSize", () => Stream.Null.CopyToAsync(Stream.Null, -123));
+            await Assert.ThrowsAsync<ObjectDisposedException>(() => Stream.Null.CopyToAsync(new NopStream()));
+            await Assert.ThrowsAsync<NotSupportedException>(() => Stream.Null.CopyToAsync(new ReadOnlyStream()));
+        }
+        
+        [Theory]
+        [MemberData(nameof(NullStream_ReadWriteData))]
+        public async static Task TestNullStream_Read(byte[] buffer, int offset, int count)
+        {
+            int read = Stream.Null.Read(buffer, offset, count);
+            Assert.Equal(0, read);
+            
+            read = await Stream.Null.ReadAsync(buffer, offset, count);
+            Assert.Equal(0, read);
+        }
+        
+        [Fact]
+        public static void TestNullStream_ReadByte()
+        {
+            int data = Stream.Null.ReadByte();
+            Assert.Equal(-1, data);
+        }
+        
+        [Theory]
+        [MemberData(nameof(NullStream_ReadWriteData))]
+        public async static Task TestNullStream_Write(byte[] buffer, int offset, int count)
+        {
+            Stream.Null.Write(buffer, offset, count);
+            await Stream.Null.WriteAsync(buffer, offset, count);
+        }
+        
+        [Theory]
+        [MemberData(nameof(NullStream_WriteByteData))]
```

Comment:
There's no reason to run this test 256 times, once for each byte.  The test isn't even checking the effect of the write (which is a nop), so this can just be a Fact that writes a single byte, doesn't matter which.